### ISSUE TITLE
Failed to install on OS X 10.7

### ISSUE THREAD

**Issue by nonamez** [mapbox/node-sqlite3#411]
```bash
MacBook-Pro-Kiril:ghost nonamez$ node --version
v0.12.0
MacBook-Pro-Kiril:ghost nonamez$ npm --version
2.5.1
```
```bash
MacBook-Pro-Kiril:ghost nonamez$ npm install sqlite3
\
> sqlite3@3.0.4 install /Users/nonamez/Sites/ghost/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
  ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3080701/sqlite3.c
  TOUCH Release/obj.target/deps/action_before_build.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite-autoconf-3080701/sqlite3.o
Release/obj/gen/sqlite-autoconf-3080701/sqlite3.c:8645:26: warning: unused
      variable 'sqlite3one' [-Wunused-const-variable]
SQLITE_PRIVATE const int sqlite3one = 1;
                         ^
1 warning generated.
  LIBTOOL-STATIC Release/sqlite3.a
  CXX(target) Release/obj.target/node_sqlite3/src/database.o
../src/database.cc:134:18: error: no matching member function for call to 'Set'
    args.This()->Set(NanNew("filename"), args[0]->ToString(), ReadOnly);
    ~~~~~~~~~~~~~^~~
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2112:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(Handle<Value> key, Handle<Value> value);
       ^
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2114:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(uint32_t index, Handle<Value> value);
       ^
../src/database.cc:135:18: error: no matching member function for call to 'Set'
    args.This()->Set(NanNew("mode"), NanNew<Integer>(mode), ReadOnly);
    ~~~~~~~~~~~~~^~~
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2112:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(Handle<Value> key, Handle<Value> value);
       ^
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2114:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(uint32_t index, Handle<Value> value);
       ^
../src/database.cc:145:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:230:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:509:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:610:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
4 warnings and 2 errors generated.
make: *** [Release/obj.target/node_sqlite3/src/database.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Darwin 13.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=sqlite3" "--configuration=Release" "--module_name=node_sqlite3" "--version=3.0.4" "--major=3" "--patch=4" "--runtime=node" "--node_abi=node-v14" "--platform=darwin" "--target_platform=darwin" "--arch=x64" "--target_arch=x64" "--module_main=./lib/sqlite3" "--host=https://mapbox-node-binary.s3.amazonaws.com/" "--module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64" "--module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node" "--remote_path=./sqlite3/v3.0.4/" "--package_name=node-v14-darwin-x64.tar.gz" "--staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz" "--hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/" "--hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz"
gyp ERR! cwd /Users/nonamez/Sites/ghost/node_modules/sqlite3
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js rebuild --name=sqlite3 --configuration=Release --module_name=node_sqlite3 --version=3.0.4 --major=3 --patch=4 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./lib/sqlite3 --host=https://mapbox-node-binary.s3.amazonaws.com/ --module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64 --module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node --remote_path=./sqlite3/v3.0.4/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz --hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/ --hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/nonamez/Sites/ghost/node_modules/sqlite3/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:1008:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1080:5)
node-pre-gyp ERR! System Darwin 13.4.0
node-pre-gyp ERR! command "node" "/Users/nonamez/Sites/ghost/node_modules/sqlite3/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/nonamez/Sites/ghost/node_modules/sqlite3
node-pre-gyp ERR! node -v v0.12.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.1
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js rebuild --name=sqlite3 --configuration=Release --module_name=node_sqlite3 --version=3.0.4 --major=3 --patch=4 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./lib/sqlite3 --host=https://mapbox-node-binary.s3.amazonaws.com/ --module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64 --module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node --remote_path=./sqlite3/v3.0.4/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz --hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/ --hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz' (1)
npm ERR! Darwin 13.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "sqlite3"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! sqlite3@3.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sqlite3@3.0.4 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the sqlite3 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls sqlite3
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/nonamez/Sites/ghost/npm-debug.log
```
```bash
MacBook-Pro-Kiril:ghost nonamez$ npm install sqlite3 --build-from-source
\
> sqlite3@3.0.4 install /Users/nonamez/Sites/ghost/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
  ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3080701/sqlite3.c
  TOUCH Release/obj.target/deps/action_before_build.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite-autoconf-3080701/sqlite3.o
Release/obj/gen/sqlite-autoconf-3080701/sqlite3.c:8645:26: warning: unused
      variable 'sqlite3one' [-Wunused-const-variable]
SQLITE_PRIVATE const int sqlite3one = 1;
                         ^
1 warning generated.
  LIBTOOL-STATIC Release/sqlite3.a
  CXX(target) Release/obj.target/node_sqlite3/src/database.o
../src/database.cc:134:18: error: no matching member function for call to 'Set'
    args.This()->Set(NanNew("filename"), args[0]->ToString(), ReadOnly);
    ~~~~~~~~~~~~~^~~
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2112:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(Handle<Value> key, Handle<Value> value);
       ^
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2114:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(uint32_t index, Handle<Value> value);
       ^
../src/database.cc:135:18: error: no matching member function for call to 'Set'
    args.This()->Set(NanNew("mode"), NanNew<Integer>(mode), ReadOnly);
    ~~~~~~~~~~~~~^~~
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2112:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(Handle<Value> key, Handle<Value> value);
       ^
/Users/nonamez/.node-gyp/0.12.0/deps/v8/include/v8.h:2114:8: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
  bool Set(uint32_t index, Handle<Value> value);
       ^
../src/database.cc:145:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:230:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:509:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
../src/database.cc:610:9: warning: unused variable 'status' [-Wunused-variable]
    int status = uv_queue_work(uv_default_loop(),
        ^
4 warnings and 2 errors generated.
make: *** [Release/obj.target/node_sqlite3/src/database.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Darwin 13.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=sqlite3" "--configuration=Release" "--module_name=node_sqlite3" "--version=3.0.4" "--major=3" "--patch=4" "--runtime=node" "--node_abi=node-v14" "--platform=darwin" "--target_platform=darwin" "--arch=x64" "--target_arch=x64" "--module_main=./lib/sqlite3" "--host=https://mapbox-node-binary.s3.amazonaws.com/" "--module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64" "--module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node" "--remote_path=./sqlite3/v3.0.4/" "--package_name=node-v14-darwin-x64.tar.gz" "--staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz" "--hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/" "--hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz"
gyp ERR! cwd /Users/nonamez/Sites/ghost/node_modules/sqlite3
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js rebuild --name=sqlite3 --configuration=Release --module_name=node_sqlite3 --version=3.0.4 --major=3 --patch=4 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./lib/sqlite3 --host=https://mapbox-node-binary.s3.amazonaws.com/ --module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64 --module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node --remote_path=./sqlite3/v3.0.4/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz --hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/ --hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/nonamez/Sites/ghost/node_modules/sqlite3/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:1008:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1080:5)
node-pre-gyp ERR! System Darwin 13.4.0
node-pre-gyp ERR! command "node" "/Users/nonamez/Sites/ghost/node_modules/sqlite3/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/nonamez/Sites/ghost/node_modules/sqlite3
node-pre-gyp ERR! node -v v0.12.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.1
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js rebuild --name=sqlite3 --configuration=Release --module_name=node_sqlite3 --version=3.0.4 --major=3 --patch=4 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./lib/sqlite3 --host=https://mapbox-node-binary.s3.amazonaws.com/ --module_path=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64 --module=/Users/nonamez/Sites/ghost/node_modules/sqlite3/lib/binding/node-v14-darwin-x64/node_sqlite3.node --remote_path=./sqlite3/v3.0.4/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz --hosted_path=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/ --hosted_tarball=https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v3.0.4/node-v14-darwin-x64.tar.gz' (1)
npm ERR! Darwin 13.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "sqlite3" "--build-from-source"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! sqlite3@3.0.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sqlite3@3.0.4 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the sqlite3 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls sqlite3
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/nonamez/Sites/ghost/npm-debug.log
```

**Comment by devanp92** [comment]
I have also got this error. I posted it [here](https://stackoverflow.com/questions/28598512/node-pre-gyp-install-fallback-to-build-error-install-with-npm-on-osx).

**Comment by springmeyer** [comment]
this is fixed in the v3.0.5 release (problem was that v3.0.4 was too early to support node v0.12.x)
